### DISCUSSION TITLE
Lots of compile issues with 2.0-SNAPSHOT

### DISCUSSION THREAD

**Discussion by guw** [eclipse-m2e/m2e-core#810] [Q&A] [answered]
I updated to m2e 2.0 latest snapshot build and I am seeing lots of compilation issues now. I recall I had to remove (uninstall) Tycho project configurators as well as some others because they weren't compatible with m2e 2.0. Is there a workaround to bring those back?

I think it's because some required source generation does not happen now.

```
Multiple markers at this line
- Plugin execution not covered by lifecycle configuration: org.codehaus.modello:modello-maven-plugin:1.8.3:xpp3-reader (execution: standard, phase: generate-sources)
- Plugin execution not covered by lifecycle configuration: org.codehaus.modello:modello-maven-plugin:1.8.3:java (execution: standard, phase: generate-sources)
- Plugin execution not covered by lifecycle configuration: org.codehaus.modello:modello-maven-plugin:1.8.3:xpp3-writer (execution: standard, phase: generate-sources)
```

Installation fails with:
```
Cannot complete the install because of a conflicting dependency.
  Software being installed: m2e connector for modello 0.16.0.201409152009 (org.sonatype.m2e.modello.feature.feature.group 0.16.0.201409152009)
  Software currently installed: M2E - Maven Integration for Eclipse (includes Incubating components) 2.0.0.20220621-0942 (org.eclipse.m2e.feature.feature.group 2.0.0.20220621-0942)
  Only one of the following can be installed at once: 
    Maven Integration for Eclipse JDT 1.5.0.20140606-0033 (org.eclipse.m2e.jdt 1.5.0.20140606-0033)
    M2E JDT Integration 1.18.3.20211112-0847 (org.eclipse.m2e.jdt 1.18.3.20211112-0847)
    M2E JDT Integration 2.0.0.20220616-1951 (org.eclipse.m2e.jdt 2.0.0.20220616-1951)
  Cannot satisfy dependency:
    From: M2E - Maven Integration for Eclipse (includes Incubating components) 2.0.0.20220621-0942 (org.eclipse.m2e.feature.feature.group 2.0.0.20220621-0942)
    To: org.eclipse.equinox.p2.iu; org.eclipse.m2e.jdt [2.0.0.20220616-1951,2.0.0.20220616-1951]
  Cannot satisfy dependency:
    From: m2e connector for modello 0.16.0.201409152009 (org.sonatype.m2e.modello 0.16.0.201409152009)
    To: osgi.bundle; org.eclipse.m2e.jdt [1.4.0,2.0.0)
  Cannot satisfy dependency:
    From: m2e connector for modello 0.16.0.201409152009 (org.sonatype.m2e.modello.feature.feature.group 0.16.0.201409152009)
    To: org.eclipse.equinox.p2.iu; org.sonatype.m2e.modello [0.16.0.201409152009,0.16.0.201409152009]
```

**Comment by HannesWell** [comment, marked as answer]
> I updated to m2e 2.0 latest snapshot build and I am seeing lots of compilation issues now. I recall I had to remove (uninstall) Tycho project configurators as well as some others because they weren't compatible with m2e 2.0. Is there a workaround to bring those back?
> 
That's right. The connectors used for development of M2E are not yet prepared for M2E 2.0.0. I thought about preparing them already but since we (mainly Christoph and Mickael) are still enhancing the API for 2.0, a lot more than just removing the previously deprecated API I think it is better to wait until this work is completed and we don't change them multiple times and maybe even in a way that will not work with the final 2.0.0. Besides that I think the Tycho Connector will not be necessary when we have embedded the right lifecycle-mappings in Tycho.
Any ways, to not have a failing setup I just have disabled the P2Director tasks to install these connectors yesterday: https://github.com/eclipse-m2e/m2e-core/pull/806



> I think it's because some required source generation does not happen now.
> 
That is very likely the reason for the failure, since the modello-connector is not installed the needed code-generation does not happen.
Running the M2E Maven build should fix that. You can use the predefined `m2e-core--build` launch for this. If you want to speed this up a bit, you can check `Skip Tests` or simply abort the build after `org.eclipse.m2e.core` has been build.

**Comment by laeubi** [comment]
Yeah please don't invest time at the Tycho stuff we should simply embedd a minimal metainfo xml into tycho and throw away for the rest (probably removing the tycho one from the cataloger) as it is heavily outdated.

Beside that, you could help the m2e project (and its users) by collecting relevant m2e connectors and either:

1. If the connector is relevant to the "core" maven-eco-system, no longer maintained but has an EPL2.0 compatible license try to get IP clearance and help adopting it for m2e
2. If the connector is relevant and actively maintained, make their maintainer aware of m2e 2.0 and maybe help them to migrate

We most likely need to find a way to reject "bad" connectors in some way (maybe by their import/exports) but that's not in the queue currently...

**Reply by guw** [reply]
I agree. The concept of extensible connectors is nice but having this in the same repository and maintained and released with m2e is a better user experience for the common Maven plug-ins. So +1 on getting them into GitHub here.
